modules/templates/templates/templates.test.tsx

```diff
@@ -0,0 +1,732 @@
+    });
+
+    it("Checks modal pops up when delete icon clicked", async () => {
+      render(<Templates />);
+
+      // Wait for the table to be fully rendered (at least one row is filled)
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getAllByTestId("templates.delete");
+      act(() => templateEdit[0].click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("templates.delete-confirmation"),
+        ).toBeInTheDocument();
+      });
+    });
+
+    it("Checks template delete function", async () => {
+      render(<Templates />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const templateEdit = screen.getAllByTestId("templates.delete");
+      act(() => templateEdit[0].click());
+
+      await waitFor(() => {
+        expect(
+          screen.getByTestId("templates.delete-confirmation"),
+        ).toBeInTheDocument();
+      });
+
+      const confirmButton = screen.getByTestId("modal.confirm-button");
+      expect(confirmButton).toBeInTheDocument();
+      act(() => confirmButton.click());
+
+      await waitFor(() => {
+        expect(
+          screen.queryByTestId("templates.delete-confirmation"),
+        ).not.toBeInTheDocument();
+      });
+
+      // TODO: Make this check dynamic
+      expect(screen.queryByText("New Template Name")).not.toBeInTheDocument();
+    });
+
+    it("Checks template redirect on select", async () => {
+      const spy = jest.spyOn(RouterMock, "push");
+      render(<Templates />);
+
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+
+      const headers = screen.getAllByTestId("templates.title-cell");
+      act(() => headers[1].click());
+
+      expect(spy).toHaveBeenCalledTimes(1);
+      expect(spy).toHaveBeenCalledWith(
+        expect.stringMatching(/\/app\/template-detail\?id=.*/),
+      );
+    });
+  });
+
+  describe("Templates Detail Page", () => {
+    let TemplateDetail: CoreComponent;
+    beforeAll(() => {
+      TemplateDetail = getComponent("app", "template-detail");
+    });
+
+    beforeEach(() => {
+      setRouteQuery({ id: "2" });
+    });
+    afterEach(() => {});
+
+    it("Templates Detail is Mounted", async () => {
+      await act(() => render(<TemplateDetail />));
+
+      expect(screen.getByTestId("template-detail.title")).toBeInTheDocument();
+      expect(
+        screen.getByTestId("template-detail.add-field"),
+      ).toBeInTheDocument();
+      expect(
+        screen.queryByText("template-detail.table"),
+      ).not.toBeInTheDocument();
+      screen.debug();
+    });
+
+    it("Checks table is populated", async () => {
+      render(<TemplateDetail />);
+
+      // Wait for the table to be fully rendered (at least one row is filled)
+      await waitFor(() => {
+        expect(screen.getByTestId("table-row-0"));
+      });
+    });
```

Comment:
Will merge with the previous test (is more clear)  no need the `not.toBeInDocument` with that will be understood needs to load first